### PR TITLE
test: adjust Frontend tests for Windows

### PR DESCRIPTION
**Pull request by compnerd**
Adjust some of the tests to improve the pass rate on Windows.  This
involves using less shell syntax, quoting instead of escaping slashes
and accepting the windows path separator.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @jrose-apple @CodaFi

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

File: test/Frontend/batch_mode_output_file_map_missing_primary.swift [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // RUN: %empty-directory(%t)
 // RUN: echo 'print("Hello, World!")' >%t/main.swift
 // RUN: touch %t/file-01.swift
-// RUN: (cd %t && not %target-swift-frontend -parse -primary-file main.swift -primary-file file-01.swift -supplementary-output-file-map %S/Inputs/supplementary_output_filemap_missing_a_primary.yaml >%t/errs.txt 2>&1)
+// RUN: cd %t
```
Comment (jrose-apple) [inline test/Frontend/batch_mode_output_file_map_missing_primary.swift]: I'm really happy this works now; it used to not.

**Review by jrose-apple — APPROVED** [review]

**Review by CodaFi — APPROVED** [review]
VFS test changes LGTM

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

**Pull request merged**